vortex-io/src/io_buf.rs

```diff
@@ -34,13 +34,13 @@ pub unsafe trait IoBuf: Unpin + 'static {
 }
 
 /// An owned view into a contiguous sequence of bytes.
-pub struct Slice<T> {
+pub struct OwnedSlice<T> {
```

Comment:
We can also just remove it, not sure why we need it right now but `Slice` is definitely already a thing.